core/templates/components/question-directives/question-editor/question-editor.component.ts

```diff
@@ -73,9 +75,7 @@ export class QuestionEditorComponent implements OnInit, OnDestroy {
   ) { }
 
   saveInteractionId(displayedValue: string): void {
```

Comment:
Why do we still need this method at all? Can the modal call a combined method only once the save happens?

Keeping this method encourages it to be used by clients, and you probably don't want that. Restrict the API so that only "safe" operations are allowed.

Reply:
Sorry for the delay on this, @yashwardhan-jyani.

I mean that the same method is called for saving the (interaction ID, customization args) tuple, and it is only called when the modal dialog is closed.

I think to fix this fully we need to do a bit of standardization between how we handle the exploration editor and question editor. What is the minimum we can do that would bring them in line and keep the API safe as mentioned here?

Reply:
I think we need to refactor that to remove the two events and have just one event instead that carries both values simultaneously. We should not have two ways to do this in the two different places, it will become hard to maintain.

What does the exploration editor use the individual events for? If you can let me know that, maybe we can figure out another way for it to get the necessary signals, but let's first figure out whether those signals are actually necessary.

Reply:
Sounds good. (Though the code snippet you point to here seems to just use regular function calls rather than events.)  

Anyway, if you're sure these aren't used for anything else, let's collapse to a single event and make sure it's handled consistently between both the exploration and question editor. Try to keep the eventual changelist format the same if possible, so you don't need to update the backend.

Also one tip: before you make changes, I suggest writing tests to ensure that the correct behaviour is "locked in". That way, you'll have a better guarantee that your refactor doesn't break anything -- we don't want to find out in the next release that some related workflow here is no longer working on prod.